src/planner/planner.cpp

```diff
@@ -154,25 +155,47 @@ std::unique_ptr<LogicalPlan> Planner::planRenameProperty(const BoundStatement& s
 
 std::unique_ptr<LogicalPlan> Planner::planCopy(
     const catalog::Catalog& catalog, const BoundStatement& statement) {
-    auto& copyCSVClause = (BoundCopy&)statement;
+    auto& boundCopyStatement = (BoundCopy&)statement;
+    if (boundCopyStatement.getCopyDescription().direction == CopyDescription::Direction::FROM) {
+        return planCopyFrom(catalog, statement);
+    } else {
+        return planCopyTo(catalog, statement);
+    }
+}
+
+std::unique_ptr<LogicalPlan> Planner::planCopyFrom(
+    const catalog::Catalog& catalog, const BoundStatement& statement) {
     auto plan = std::make_unique<LogicalPlan>();
+    auto& boundCopyStatement = (BoundCopy&)statement;
     expression_vector arrowColumnExpressions;
     for (auto& property :
-        catalog.getReadOnlyVersion()->getTableSchema(copyCSVClause.getTableID())->properties) {
+        catalog.getReadOnlyVersion()->getTableSchema(boundCopyStatement.getTableID())->properties) {
         if (property.dataType.getLogicalTypeID() != common::LogicalTypeID::SERIAL) {
             arrowColumnExpressions.push_back(std::make_shared<VariableExpression>(
                 common::LogicalType{common::LogicalTypeID::ARROW_COLUMN}, property.name,
                 property.name));
         }
     }
-    auto copyCSV = make_shared<LogicalCopy>(copyCSVClause.getCopyDescription(),
-        copyCSVClause.getTableID(), copyCSVClause.getTableName(), std::move(arrowColumnExpressions),
+    auto copyCSV = make_shared<LogicalCopy>(boundCopyStatement.getCopyDescription(),
+        boundCopyStatement.getTableID(), boundCopyStatement.getTableName(),
+        std::move(arrowColumnExpressions),
         std::make_shared<VariableExpression>(
             common::LogicalType{common::LogicalTypeID::INT64}, "startOffset", "startOffset"),
-        copyCSVClause.getStatementResult()->getSingleExpressionToCollect());
+        boundCopyStatement.getStatementResult()->getSingleExpressionToCollect());
     plan->setLastOperator(std::move(copyCSV));
     return plan;
 }
 
+std::unique_ptr<LogicalPlan> Planner::planCopyTo(
+    const catalog::Catalog& catalog, const BoundStatement& statement) {
+    auto plan = std::make_unique<LogicalPlan>();
+    auto& boundCopyStatement = (BoundCopy&)statement;
+    auto copyTo = make_shared<LogicalCopyTo>(boundCopyStatement.getCopyDescription(),
```

Comment:
If your `boundCopyStatement` has a `nodeExpression`, you can call `joinOrderEnumerator.appendScanNodeID` as the first operator and `queryPlanner.appendScanNodePropIfNecessary` as the second operator. And finally a `CopyTo` on top of it.